src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -0,0 +1,279 @@
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.LanguagePreference;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.UserCategory;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.UserProfileCreationRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.request.UserTypeRequest;
+import uk.gov.hmcts.reform.cwrdapi.controllers.response.CaseWorkerProfileCreationResponse;
+import uk.gov.hmcts.reform.cwrdapi.controllers.response.UserProfileCreationResponse;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerIdamRoleAssociation;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerLocation;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerProfile;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerRole;
+import uk.gov.hmcts.reform.cwrdapi.domain.CaseWorkerWorkArea;
+import uk.gov.hmcts.reform.cwrdapi.domain.RoleType;
+import uk.gov.hmcts.reform.cwrdapi.domain.UserType;
+import uk.gov.hmcts.reform.cwrdapi.repository.CaseWorkerIdamRoleAssociationRepository;
+import uk.gov.hmcts.reform.cwrdapi.repository.CaseWorkerProfileRepository;
+import uk.gov.hmcts.reform.cwrdapi.repository.RoleTypeRepository;
+import uk.gov.hmcts.reform.cwrdapi.repository.UserTypeRepository;
+import uk.gov.hmcts.reform.cwrdapi.service.CaseWorkerService;
+import uk.gov.hmcts.reform.cwrdapi.util.JsonFeignResponseUtil;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+@Slf4j
+@Setter
+public class CaseWorkerServiceImpl implements CaseWorkerService {
+
+
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
+    @Autowired
+    CaseWorkerProfileRepository caseWorkerProfileRepo;
+
+    @Autowired
+    RoleTypeRepository roleTypeRepository;
+
+    @Autowired
+    UserTypeRepository userTypeRepository;
+
+    @Autowired
+    CaseWorkerIdamRoleAssociationRepository cwIdamRoleAssocRepository;
+
+    @Autowired
+    private UserProfileFeignClient userProfileFeignClient;
+
+    List<RoleType> roleTypes = new ArrayList<>();
+
+    List<UserType> userTypes = new ArrayList<>();
+
+    CaseWorkerProfile caseWorkerProfile = null;
+
+
+    @Override
+    public ResponseEntity<Object> saveOrUpdateOrDeleteCaseWorkerUserProfiles(List<CaseWorkersProfileCreationRequest>
+                                                                           cwrsProfilesCreationRequest) {
+        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        try {
+            getRolesAndUserTypes();
+            cwrsProfilesCreationRequest.forEach(cwrProfileCreationRequest -> {
+                CaseWorkerProfile caseWorkerProfileRespose = null;
+
+                caseWorkerProfile = caseWorkerProfileRepo
+                        .findByEmailId(cwrProfileCreationRequest.getEmailId());
```

Comment:
We can refactor this code with lamda filters.

Reply:
This method deals  different objects and various conditions so unable to implement lamda filters here.